service/worker/archiver/util.go

```diff
@@ -99,13 +97,3 @@ func tagLoggerWithActivityInfo(logger log.Logger, activityInfo activity.Info) lo
 		tag.WorkflowRunID(activityInfo.WorkflowExecution.RunID),
 		tag.Attempt(activityInfo.Attempt))
 }
-
-func convertSearchAttributesToString(searchAttr map[string]*commonpb.Payload) map[string]string {
-	searchAttrStr := make(map[string]string)
-	for k, v := range searchAttr {
-		var s string
-		_ = payload.Decode(v, &s)
-		searchAttrStr[k] = s
```

Comment:
This code makes me think that visibility archival is not actually used because it worked only for `string` search attributes and will archive empty string for all other types.